membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -74,9 +74,10 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
           if(userHasDigiSub(subscriptions)) Future.successful(\/.right[String, List[GiftSubscriptionsFromIdentityIdRecord]](Nil))
           else giftSubscriptionsForIdentityId(identityId)
         )
-        maybeGiftAttributes = giftSubscriptions
-          .sortWith((first, second) => first.TermEndDate.isAfter(second.TermEndDate)).headOption
-          .map(record => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(record.TermEndDate)))
+        maybeGiftAttributes = Option(giftSubscriptions)
+          .filter(_.nonEmpty)
+          .map(_.maxBy(_.TermEndDate.toDateTimeAtStartOfDay.getMillis).TermEndDate)
+          .map(date => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(date)))
```

Comment:
great, nice idea using map/filter rather than an if statement.  much easier to follow than before

Option converts null to None, so it suggests a Java->scala conversion.
Let's do the term end date once up front?
as follows:
```suggestion
        maybeGiftAttributes = Some(giftSubscriptions.map(_.TermEndDate))
          .filter(_.nonEmpty)
          .map(_.maxBy(_.toDateTimeAtStartOfDay.getMillis))
          .map(date => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(date)))
```